### PR TITLE
Implemented WriteHWRegsWithMask for GSP

### PR DESCRIPTION
**Pull request by kevinhartman**
Implemented WriteHWRegsWithMask for GSP.

### CODE REVIEW

File: src/core/hle/service/gsp_gpu.cpp [hunk 1]
```diff
@@ -50,20 +50,43 @@ static inline InterruptRelayQueue* GetInterruptRelayQueue(u32 thread_id) {
...
-static void WriteHWRegs(u32 base_address, u32 size_in_bytes, const u32* data) {
+/**
+ * Checks if the parameters in a register write call are valid and logs in the case that
+ * they are not
+ * @param base_address The first address in the sequence of registers that will be written
+ * @param size_in_bytes The number of registers that will be written
+ * @return true if the parameters are valid, false otherwise
+ */
+static bool CheckWriteParameters(u32 base_address, u32 size_in_bytes)
+{
```
Comment (neobrain) [inline src/core/hle/service/gsp_gpu.cpp]: `{` should be on previous line

File: src/core/hle/service/gsp_gpu.cpp [hunk 2]
```diff
@@ -50,20 +50,43 @@ static inline InterruptRelayQueue* GetInterruptRelayQueue(u32 thread_id) {
...
         LOG_ERROR(Service_GSP, "Invalid size 0x%08x", size_in_bytes);
-        return;
+        return false;
     }
 
+    return true;
+}
+
+/**
+ * Writes sequential GSP CPU hardware registers using an array of source data
```
Comment (neobrain) [inline src/core/hle/service/gsp_gpu.cpp]: "CPU"?

**Comment by neobrain** [comment]
LGTM

**Comment by kevinhartman** [comment]
Addressed review comments

**Comment by bunnei** [comment]
LGTM. Confirmed that @neobrain's comments were addressed.

**Pull request merged**